CMake/FollyConfigChecks.cmake

```diff
@@ -20,7 +20,11 @@ include(CheckSymbolExists)
 include(CheckTypeSize)
 include(CheckCXXCompilerFlag)
 
-CHECK_INCLUDE_FILE_CXX(jemalloc/jemalloc.h FOLLY_USE_JEMALLOC)
+if (NOT CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
+  CHECK_INCLUDE_FILE_CXX(jemalloc/jemalloc.h FOLLY_USE_JEMALLOC)
+else()
+  CHECK_INCLUDE_FILE_CXX(malloc_np.h FOLLY_USE_JEMALLOC)
+endif()
```

Comment:
Should switch the conditions around. That way, further conditions can easily be added as necessary.

```
if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
  ...
else()
  ...
endif()
```